### PR TITLE
Make the version of rubygems configurable

### PR DESCRIPTION
**Pull request by maxim**
This lets you set desired version of rubygems regardless of which one is installed by the brightbox package.

Slightly off-topic: I used the new [chef-dk](http://www.getchef.com/downloads/chef-dk/) and there was an error trying to run `berks`, due to the hack in the metadata.rb file where it uses `Chef::Version`. The chef's version in chef-dk is actually 11.14.0.alpha.1 and cannot be parsed by `Chef::Version`. So I had to comment out all those attribute declarations to get specs to run (and add another spec). Regardless, that junk in metadata.rb should be removed, I will submit a separate pull request for that.

### CODE REVIEW

**Comment by benlangfeld** [comment]
I would appreciate a fix for "that junk"; I'm interested to know what your solution is.

File: README.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ For more documentation on these builds, please see [the official brightbox docum
 * `node['brightbox-ruby']['version']` - The version of Ruby to install. Defaults to `2.1`.
 * `node['brightbox-ruby']['install_dev_package']` - Install the dev package, which provides headers for gem native extensions. Available options: `true`, `false`. Defaults to `true`.
 * `node['brightbox-ruby']['gems']` - Gems to be installed by default. Defaults to `["bundler", "rake", "rubygems-bundler"]`.
+* `node['brightbox-ruby']['rubygems_version']` - The version of rubygems to enforce. Set to nil to use the default version packaged by brightbox. Defaults to `nil`.
```
Comment (benlangfeld) [inline README.md]: This content needs to be included in `metadata.rb`, since these docs are generated by `make docs`.

**Comment by benlangfeld** [comment]
Thanks for the contribution. Ignore @houndci, it's being excessive. All tests pass fine for me, my only complaint is the docs mis-match. Could you fix that and run `make docs` to generate the docs properly? Once that's done I'll gladly merge.

**Comment by maxim** [comment]
@benlangfeld oh apologies for the junk reference, I didn't realize it was recently added, simply assumed it was old. Do you use the attribute declarations in metadata? The conversation I had in #chef-hacking irc made me believe they are no longer useful, and myself I never had need for them. If you are using them, I'll find a way to get rid of chef dependency. I try to keep metadata file as simple/declarative as possible, and `Chef::Version` is definitely not going to work, since that class only supports `x.y(.z)` version format, and chef itself can have anything there.

**Comment by benlangfeld** [comment]
Including attribute documentation in `metadata.rb` makes it machine-readable. We use it to generate the README, and it can be used by other tools to better annotate lists of attributes. I'd like to hear more about it being "no longer useful" - I think the main problem is simply that most cookbooks don't include it, when really they should.

**Comment by maxim** [comment]
Looks like it was just a person's opinion. I do agree however that metadata should remain dumb if possible. Either way, let me see what I can come up with.

**Comment by benlangfeld** [comment]
Definitely prefer metadata.rb to be dumb, especially since it's compiled to JSON for packaging. The logic in there right now was a last resort in response to an undocumented API change in Chef.

**Comment by maxim** [comment]
@benlangfeld should be good now, I'll open another PR when I find a good Chef::Version solution

**Comment by benlangfeld** [comment]
I've gone ahead and removed that logic in 0359c824202e6cc95801fc50b0a1565d26fbbb93. I'll release 1.1.0 shortly. Let me know if you have any problems.

**Comment by maxim** [comment]
Don't forget about the require chef line on top.

**Comment by benlangfeld** [comment]
Damnit, I did. I don't think that'll affect the release, though, so I'll just remove it on the develop branch and it can slip in next time.

**Pull request merged**